### ISSUE TITLE
Cannot Connect

### ISSUE THREAD

**Issue by fabltd** [DVSwitch/IAX_Bridge#2]
Hi

Have the bridge running. Node shows in Allstar Link. When trying to connect the connection is rejected with: Cloud not negotiate common codec.

**Comment by MrBungle42** [comment]
IAX_Bridge supports alaw and ulaw right now

**Comment by fabltd** [comment]
Should be good. I am using the repeater phone app on IOS to connect to Allstar.

It supports - G.711 uLAW  assum this is correct.

Should I see anything in the logs for a connection request. It appears to be rejected.

If you are able to try its MB7IJT 596601

**Comment by fabltd** [comment]
I am unable to connect from anther client either. Transive on the mac.

I don't see any connection attempts in the log. Are these logged.

Removing port 4569 in the firewall cause the app to keep dialing. Puting it back it returns with the codec error. 

So I think traffic is getting thought but is there some kind of authentication issue?

**Comment by fabltd** [comment]
@MrBungle42 whats the largest verbose for the debug? I am spoke to the repeater phone all dev. He seem to think the codec support is not being properly presented?

**Comment by fabltd** [comment]
A TCP Dump is showing: 

 0x0020:  0000 0000 0000 0610 1612 5061 7373 776f  ..........Passwo
        0x0030:  7264 2069 6e63 6f72 7265 6374 2a01 1d    rd.incorrect*..

Where is the password set?

What passoword is incorrect ? 

The password used to connect to ASL is correct and the bridge shows as being conncted?

Is this meant to work or do I need to revert to a full asterisk install?

**Comment by N4IRS** [comment]
Yes, it is meant to work. Running on node 414983 now. With web console at <https://server.dvswitch.org> making outbound and accepting inbound connection (one at a time).

**Comment by fabltd** [comment]
Hi I tried connecting to that node but it does not answer for me.

**Comment by N4IRS** [comment]
What node number / type are you connecting from?

**Comment by fabltd** [comment]
Using the Repeater Phone app on IOS. M1CVZ

**Comment by N4IRS** [comment]
Is that really a TCP dump or a UDP dump? What port?

**Comment by fabltd** [comment]
> Is that really a TCP dump or a UDP dump? What port?

4569 used tcpdump but is udp
Port.

**Comment by N4IRS** [comment]
The string "incorrect" does not exist in IAX_Bridge

**Comment by fabltd** [comment]
> The string "incorrect" does not exist in IAX_Bridge
> 
> 

It's coming back from the repeater phone app. Some where between bridge and the app the codec is not being agreed. I just cannot debug where

**Comment by fabltd** [comment]
So it seams there is an issue in the implementation of IAX2.

The call initialization handshake never even gets to the point of negotiating codecs; it is failing when requesting a calltoken.

It must be that the DVSwitch IAX stack is older than the current versions of AllStarLink and HamVOIP that are in operation today. The [calltoken extension was proposed back in 2009](https://docs.asterisk.org/Configuration/Channel-Drivers/Inter-Asterisk-eXchange-protocol-version-2-IAX2/IAX2-Security/?h=). 

Basically, a calltoken exchange should look like this:
Client sends a NEW packet, with an empty calltoken IE (Information Element)
Server responds with a CALLTOKEN packet, containing a randomly generated token
Client re-sends the NEW request, with the calltoken and the additional details, such as preferred codec
Call setup proceeds as normal

Can this be addressed or shall I revert to asterisk?

**Comment by w8wjb** [comment]
Hi all, RepeaterPhone developer here. Jeremy sent me a packet capture and I analyzed it; those are the findings that he posted.

As I told him, I can potentially make some changes to the IAX stack in RepeaterPhone that could potentially make things work, but other clients, such as Transceive may still have problems connecting.

What IAX stack are you using?

**Comment by N4IRS** [comment]
Wes,
Thanks for reaching out. We are looking at IAX_Bridge in light of some internal testing. I'll keep you informed. You can reach me direct at n4irs@dvswitch.org

73, Steve N4IRS

**Comment by N4IRS** [comment]
We are testing a update to IAX_Bridge on node 414983. The web console is at <https://server.dvswitch.org>

**Comment by fabltd** [comment]
Hi

Tried to connect but it appears it's offline ?

**Comment by N4IRS** [comment]
No, it's on-line

**Comment by fabltd** [comment]
Cannot connect from the repeater phone app. The call is not answered.

![image](https://github.com/DVSwitch/IAX_Bridge/assets/4989766/0122b1b6-0ca3-4dfc-bb1f-121a1f56bd33)

**Comment by N4IRS** [comment]
Try again

**Comment by fabltd** [comment]
Getting closer. Part connects but not fully ![image](https://github.com/DVSwitch/IAX_Bridge/assets/4989766/c489c0d7-2e38-44cb-ad8d-28c884d22c60)

**Comment by N4IRS** [comment]
What node number did you assign to your app? I see this:
Connect from node **414983** at address 2.24.2XX.XXX failed.

**Comment by fabltd** [comment]
Hi

Yes that's me.

**Comment by N4IRS** [comment]
What node number did you assign to your app?

**Comment by fabltd** [comment]
Hi

Didn't just signed in with user name and password.

**Comment by N4IRS** [comment]
Connect to node 42564

**Comment by fabltd** [comment]
That works

**Comment by N4IRS** [comment]
I will look at the logs when I get a chance

**Comment by N4IRS** [comment]
RepeaterPhone connects as AllStar-Public. IAX_Bridge ONLY accepts a node connection. I am closing this issue.

**Comment by w8wjb** [comment]
So, that isn't completely accurate. It is possible to connect with IAX username/password credentials as well. However it does not currently register a node number.

**Comment by N4IRS** [comment]
Wes,
Sorry, I was being simplistic. IAX_Bridge only accepts a node connection. Thanks for your help in this issue.
73, Steve N4IRS

**Comment by fabltd** [comment]
So there is no resolution ? @w8wjb is this something you can resolve ?

**Comment by N4IRS** [comment]
Jeremy,
IAX_Bridge is intended to connect DVSwitch Server to the AllStar network. The use case for accepting a inbound connection from RepeaterPhone or DVSwitch Mobile is not what IB is used for.
